logback-core/src/main/java/ch/qos/logback/core/joran/conditional/PropertyEvalJavaScriptBuilder.java

```diff
@@ -0,0 +1,94 @@
+package ch.qos.logback.core.joran.conditional;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.lang.ref.WeakReference;
+
+import javax.script.Bindings;
+import javax.script.ScriptContext;
+import javax.script.ScriptEngine;
+import javax.script.ScriptEngineManager;
+import javax.script.ScriptException;
+
+import ch.qos.logback.core.Context;
+import ch.qos.logback.core.spi.PropertyContainer;
+
+public class PropertyEvalJavaScriptBuilder implements Evaluation {
+
+    public static final String CONDITIONAL_JS_ATTR = "conditional-js";
+    private static final String JS_ENGINE_NAME = "JavaScript";
+
+    // variable names provided in JavaScript context
+    static final String PROPERTIES = "props";
+    static final String CONTEXT = "ctx";
+
+    private static WeakReference<ScriptEngine> globalEngine = new WeakReference<ScriptEngine>(null);
+    private Context context;
+    ScriptEngine engine;
```

Comment:
Yes, I wanted to check that different "if" conditions re-use the same engine. I found logback to be somewhat liberal with the scope of some member variables, so I followed along.